google-cloud-bigquery/lib/google/cloud/bigquery/routine/list.rb

```diff
@@ -0,0 +1,164 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "delegate"
+
+module Google
+  module Cloud
+    module Bigquery
+      class Routine
+        ##
+        # Routine::List is a special case Array with additional values.
+        class List < DelegateClass(::Array)
```

Comment:
It looks like this is missing support for supported list params.  Added a comment to the impl guide because I realized the docs weren't properly exposing their presence.